Docs/source/developers/profiling.rst

```diff
@@ -208,3 +212,80 @@ In these examples, the individual lines for recording a trace profile are:
 Now open the created trace files (per rank) in the Nsight-Systems GUI.
 This can be done on another system than the one that recorded the traces.
 For example, if you record on a cluster and open the analysis GUI on your laptop, it is recommended to make sure that versions of Nsight-Systems match on the remote and local system.
+
+Nvidia Nsight-Compute
+---------------------
+
+`Vendor homepage <https://developer.nvidia.com/nsight-compute>`__ and `product manual <https://docs.nvidia.com/nsight-compute/>`__.
+
+Nsight-Compute captures fine grained information at the kernel level
+concerning resource utilization. It collects a lot of data, and runs slowly (can
+be a few minutes per step), but provides detailed information about occupancy,
+and memory bandwidth for a kernel.
+
+
+Example
+""""""""""""""""""
+
+Example of how to create traces on a single-GPU system. A jobscript for
+Perlmutter is shown, but the `SBATCH` headers are not strictly necessary as the
+command only profiles a single process. This can also be run on an interactive
+node, or without a workload management system.
+
+.. code-block:: bash
+
+   #!/bin/bash -l
+   #SBATCH -t 00:30:00
+   #SBATCH -N 1
+   #SBATCH -J ncuProfiling
+   #SBATCH -A <your account>
+   #SBATCH -q regular
+   #SBATCH -C gpu
+   #SBATCH --ntasks-per-node=1
+   #SBATCH --gpus-per-task=1
+   #SBATCH --gpu-bind=map_gpu:0
+   #SBATCH --mail-user=<email>
+   #SBATCH --mail-type=ALL
+
+   # record
+   dcgmi profile --pause
+   ncu -f -o out \
+   --target-processes all \
+   --set detailed \
+   --nvtx --nvtx-include="WarpXParticleContainer::DepositCurrent::CurrentDeposition/" \
+   ./warpx input max_step=1 \
+   &> warpxOut.txt
+
+.. note::
+
+    For those used to Nsight-Systems, Nsight-Compute runs much slower. A
+    single process can take a few minutes for every step, but since we're
+    profiling at the kernel level, a single step of a single process is often
+    enough to extract relevant information.
+
+Details
+"""""""
+In the example above, the individual lines for recording a trace profile are:
+
+* ``dcgmi profile --pause`` other profiling tools can't be collecting data,
+  `see this Q&A <https://forums.developer.nvidia.com/t/profiling-failed-because-a-driver-resource-was-unavailable/205435>`_.
```

Comment:
Anonymous link might be better here
```suggestion
  `see this Q&A <https://forums.developer.nvidia.com/t/profiling-failed-because-a-driver-resource-was-unavailable/205435>`__.
```